pyproject.toml

```diff
@@ -372,10 +372,14 @@ lint.ignore = [  # NOTE: non-permanent exclusions should be added to `.ruff.toml
     # pandas-vet (PD)
     "PD",
 
-    # pylint (PLR and PLW)
+    # pylint (PLC, PLR and PLW)
+    "PLC0415", # import-outside-top-level
     "PLR1730", # if-stmt-min-max (not always clearer, and sometimes slower)
     "PLW0642", # self-or-cls-assignment (occasionally desirable, very rarely a mistake)
 
+    # flake8-pytest-style
+    "PT031", # pytest-warns-with-multiple-statements
```

Comment:
Why should this rule be ignored? I can't think of any good reasons.

Reply:
Refactoring 68 tests seems too high of a risk and too much effort for a small gain in purity.

Reply:
It would be fine to have temporary ignores in `.ruff.toml`, preferably in the sub-package sections.